src/bigtiffimage.cpp

```diff
@@ -103,20 +105,31 @@ namespace Exiv2
                 byte buffer[4];
                 io.read(buffer, 4);
 
+                if (read < 4)
+                    throw Exiv2::Error(58);
+
                 const uint32_t offset = getULong(buffer, byteOrder);
                 result = Header(byteOrder, magic, 4, offset);
             }
             else
             {
                 byte buffer[8] = {0, 0, 0, 0, 0, 0, 0, 0};
                 io.read(buffer, 2);
```

Comment:
Should be `read = io.read(buffer, 2);`

Reply:
Dan:  I thought you were going to add an enum in error.hpp for errors codes.  Something like kerInsufficientMemory = 58.

Reply:
Didn't have the time to add the error codes yet, sorry. That issue has been dormant in #109 for quite some time now unfortunately.

Reply:
@clanmills @D4N take a look here: http://bigtiff.org/#FILE_FORMAT after MAGIC there are 2 bytes for big tiff.